vault/audit.go

```diff
@@ -390,22 +390,21 @@ func (c *Core) persistAudit(ctx context.Context, table *MountTable, localOnly bo
 // setupAudit is invoked after we've loaded the audit able to
 // initialize the audit backends
 func (c *Core) setupAudits(ctx context.Context) error {
-	brokerLogger := c.baseLogger.Named("audit")
-	c.AddLogger(brokerLogger)
+	c.auditLock.Lock()
+	defer c.auditLock.Unlock()
```

Comment:
The locking has been moved up to the top of the method now.

Adding the logger to `Core` now only happens just before we return on the happy path.